Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/ImpendingTest.java

```diff
@@ -0,0 +1,226 @@
+package org.mage.test.cards.abilities.keywords;
+
+import mage.abilities.keyword.ImpendingAbility;
+import mage.constants.CardType;
+import mage.constants.PhaseStep;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.permanent.Permanent;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author TheElk801
+ */
+public class ImpendingTest extends CardTestPlayerBase {
+
+    private static final String hauntwoods = "Overlord of the Hauntwoods";
+
+    public void assertHasImpending(String name, boolean hasAbility) {
+        Permanent permanent = getPermanent(hauntwoods);
```

Comment:
probably should use the name param rather than hardcoding

Reply:
whoops good catch thanks